sdk/iOS/src/MSSyncContext.h

```diff
@@ -54,19 +54,19 @@
 /// @{
 
 /// Returns a dictionary containing the items and totalCount
-- (MSSyncContextReadResult *) readWithQuery:(MSQuery *)query orError:(NSError **)error;
+- (nullable MSSyncContextReadResult *) readWithQuery:(nullable MSQuery *)query orError:(NSError * __nullable * __nullable)error;
 
 /// Should retrieve a single item from the local store or nil if item with the given ID does not exist.
--(NSDictionary *) readTable:(NSString *)table withItemId:(NSString *)itemId orError:(NSError **)error;
+-(nullable NSDictionary *) readTable:(nonnull NSString *)table withItemId:(nonnull NSString *)itemId orError:(NSError * __nullable * __nullable)error;
 
 /// Should insert/update the given item in the local store as appropriate
--(BOOL) upsertItems:(NSArray *)item table:(NSString *)table orError:(NSError **)error;
+-(BOOL) upsertItems:(nullable NSArray *)item table:(nonnull NSString *)table orError:(NSError * __nullable * __nullable)error;
 
 /// Should remove the provided item from the local store
--(BOOL) deleteItemsWithIds:(NSArray *)items table:(NSString *)table orError:(NSError **)error;
+-(BOOL) deleteItemsWithIds:(nullable NSArray *)items table:(nonnull NSString *)table orError:(NSError * __nullable * __nullable)error;
 
 /// Should remove all entries from the specified table in the local store
--(BOOL) deleteUsingQuery:(MSQuery *)query orError:(NSError **)error;
+-(BOOL) deleteUsingQuery:(nullable MSQuery *)query orError:(NSError * __nullable * __nullable)error;
```

Comment:
Well the query is used to construct the NSFetchRequest. If it or all it's properties are nil then the predicate, sort etc of the fetch request will just be nil and hence I assumed the query could be nullable.